src/openms/include/OpenMS/FILTERING/CALIBRATION/InternalCalibration.h

```diff
@@ -70,402 +70,362 @@ namespace OpenMS
+      The calibration function is calculated for each spectrum
+      separately. If not enough reference masses are found for a spectrum it is left uncalibrated.
+      For the matching of the reference masses and the peaks, the parameter mz_tolerance is used to
+      calculate a window around the reference masses. If more than one peak is found within this window the
+      closest peak is taken.
 
-         The calibration function is calculated for the whole map.
-     For the matching of the reference masses and the peaks the parameter mz_tolerance is used to
-     calculate a window around the reference masses. If more than one peak is found within this window the
-     closest peak is taken.
-     @param exp the uncalibrated peak map
-     @param calibrated_exp the calibrated peak map
-     @param ref_masses the reference m/z values
-     @param trafo_file_name file where the transformation function of the calibration is stored
+      @param exp The peak map to calibrate
+      @param ref_masses The reference m/z values
     */
     template <typename InputPeakType>
-    void calibrateMapGlobally(const MSExperiment<InputPeakType> & exp, MSExperiment<InputPeakType> & calibrated_exp, std::vector<double> & ref_masses, String trafo_file_name = "");
+    void calibrateMapSpectrumwise(MSExperiment<InputPeakType>& exp, std::vector<double>& ref_masses)
+    {
+#ifdef DEBUG_CALIBRATION
+      std::cout.precision(writtenDigits<double>(0.0));
+#endif
+      if (exp.empty())
+      {
+        std::cout << "Input is empty." << std::endl;
+        return;
+      }
 
-    /**
-     @brief Calibrate a peak map using given reference ids with one calibration function for the whole map.
+      if (exp[0].getType() != SpectrumSettings::PEAKS)
+      {
+        std::cout << "Attention: this function is assuming peak data." << std::endl;
+      }
 
-         Calibrate a map using given peptide identifications. The calibration function is calculated for the whole map.
-     The m/z-values of the reference identifications are calculated through the given sequence and charge of the peptide.
-     For the matching of the reference masses and the peaks the parameter mz_tolerance is used to
-     calculate a window around the reference masses. If more than one peak is found within this window the
-     closest peak is taken.
-     @param exp the uncalibrated peak map
-     @param calibrated_exp the calibrated peak map
-     @param ref_ids the reference peptide identifications
-     @param trafo_file_name file where the transformation function of the calibration is stored
-    */
-    template <typename InputPeakType>
-    void calibrateMapGlobally(const MSExperiment<InputPeakType> & exp, MSExperiment<InputPeakType> & calibrated_exp, std::vector<PeptideIdentification> & ref_ids, String trafo_file_name = "");
+      Size num_ref_peaks = ref_masses.size();
+      bool use_ppm = (param_.getValue("mz_tolerance_unit") == "ppm") ? true : false;
```

Comment:
I think it is "cleaner" and faster if they are members, also more consistent with other algorithms. But I dont think it is really a requirement